compiler/expressions.go

```diff
@@ -1112,6 +1112,12 @@ func (c *funcContext) translateImplicitConversion(expr ast.Expr, desiredType typ
 	}
 
 	exprType := c.p.TypeOf(expr)
+
+	// whenever expr has an underlying *types.Interface type we need to $copyInterfaceVal
+	if _, isIface := exprType.Underlying().(*types.Interface); isIface {
```

Comment:
I don't think this is the right place for this code. As far as I can see, the only place it is needed is for the receiver when calling a method on an interface (your test case). Thus, this code should be at a place where it only handles the receiver. I even think that when moving the code there, you can get rid of the `$copyInterfaceVal` helper.